### PR TITLE
Fix default product tax rate in Dashboard 1.0

### LINKED ISSUE #2953: Product form tax_rate
### What I'm trying to achieve
When I edit a product on the dashboard I would like to have the product tax_rate as the selected option. Instead the product type tax_rate is selected.

### Steps to reproduce the problem
1. Set a tax_rate on a product type
2. Set a different tax rate on a product
3. Go to the dashboard and edit the product
4. When you look at the tax_rate select field, the selected value is the product type tax_rate 

### What I expected to happen
When I edit a product I would like to have the product tax_rate (if set) as the selected option.

### Screenshots
issue_saleor

**System information**
Operating system: OSX 10.13
Browser: Firefox

Thank you.

### PR DESCRIPTION
**Pull request by tomcio1205**
I want to merge this change because it resolve #2953 

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.

### CODE REVIEW

**Comment by tomcio1205** [comment]
@Pacu2 I'm not sure how to add placeholder in select field. Should I add next option with product type tax rate but with empty value and set it as default when no tax rate is assigned on the Product ?

**Comment by Pacu2** [comment]
I think we could change the empty value description.

Let's say:
If there's no tax_rate defined on ProductType we should display `--------` as it is at this moment. 
In other cases, we could display something like (greyed out) `VAT 5% (Inherited from ProductType)` (or something shorter)

**Comment by patrys** [comment]
If we're going to use the default value then I suggest it says something like "Controlled by product type (currently: ____)".

BTW, do we ever set explicit tax percentage (eg. 5%) instead of tax groups ("hospitality")?

**Comment by Pacu2** [comment]
We are always using tax groups, `VAT 5%` was made up by me

File: saleor/product/models.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ class ProductType(models.Model):
         'Attribute', related_name='product_variant_types', blank=True)
     is_shipping_required = models.BooleanField(default=False)
     tax_rate = models.CharField(
-        max_length=128, default=DEFAULT_TAX_RATE_NAME, blank=True,
+        max_length=128, blank=True,
```
Comment (maarcingebala) [inline saleor/product/models.py]: I think that using `DEFAULT_TAX_RATE_NAME` here and in the Product model is not problematic and we could keep it. What needs to be fixed is only the dashboard form.

Comment (maarcingebala) [inline saleor/product/models.py]: I think Django generates migrations for such changes, if yes, we should provide them with in PR.

File: saleor/dashboard/product/forms.py [hunk 2]
```diff
@@ -241,7 +241,8 @@ class Meta:
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         product_type = self.instance.product_type
-        self.initial['tax_rate'] = product_type.tax_rate
+        if self.instance.tax_rate:
```
Comment (maarcingebala) [inline saleor/dashboard/product/forms.py]: If `instance.tax_rate` is None we want to render the default from the product type. I'd suggest:

```
self.initial['tax_rate'] = self.instance.tax_rate or product_type.tax_rate
```

**Comment by maarcingebala** [comment]
@tomcio1205 Would you have time to take a look at this PR again? It requires a few little changes and it would be ready to merge.

**Comment by maarcingebala** [comment]
@tomcio1205 We'd need to rebase it with `master` as well.

**Comment by maarcingebala** [comment]
Closing in favor of #3880.

**Pull request closed without merging**